internal/server/iigointernal/judiciary.go

```diff
@@ -0,0 +1,196 @@
+	rulesAffectedBySpeaker := j.EvaluationResults[j.speakerID]
+	indexOfBallotRule, err := searchForRule("inspect_ballot_rule", rulesAffectedBySpeaker.Rules)
+	if err == nil {
+		return rulesAffectedBySpeaker.Evaluations[indexOfBallotRule], nil
+	} else {
+		return true, errors.Errorf("Speaker did not conduct any ballots")
+	}
+}
+
+// inspectAllocation checks each resource allocation action adheres to the rules
+func (j *judiciary) inspectAllocation() (bool, error) {
+	// 1. Evaluate difference between commonPoolNew and commonPoolOld
+	//    to check resource allocation changes are in line with ResourceRequests
+	//    in previous resourceAllocation
+	// 2. Compare each resource allocation action adheres to rules in ruleSet
+	//    matrix
+	j.budget -= serviceCharge // will be removed post-MVP
+	rulesAffectedByPresident := j.EvaluationResults[j.presidentID]
+	indexOfAllocRule, err := searchForRule("inspect_allocation_rule", rulesAffectedByPresident.Rules)
+	if err == nil {
+		return rulesAffectedByPresident.Evaluations[indexOfAllocRule], nil
+	} else {
+		return true, errors.Errorf("President didn't conduct any allocations")
+	}
+}
+
+// searchForRule searches for a given rule in the RuleMatrix
+func searchForRule(ruleName string, listOfRuleMatrices []rules.RuleMatrix) (int, error) {
+	for i, v := range listOfRuleMatrices {
+		if v.RuleName == ruleName {
+			return i, nil
+		}
+	}
+	return 0, errors.Errorf("The rule name '%v' was not found", ruleName)
+}
+
+// declareSpeakerPerformanceWrapped wraps the result of DeclareSpeakerPerformance for orchestration
+func (j *judiciary) declareSpeakerPerformanceWrapped() {
+
+	result, checkRole := j.clientJudge.DeclareSpeakerPerformance()
+
+	message := generateSpeakerPerformanceMessage(j.BallotID, result, j.speakerID, checkRole)
+	broadcastToAllIslands(shared.TeamIDs[j.JudgeID], message)
+
+}
+
+// declarePresidentPerformanceWrapped wraps the result of DeclarePresidentPerformance for orchestration
+func (j *judiciary) declarePresidentPerformanceWrapped() {
+
+	result, checkRole := j.clientJudge.DeclarePresidentPerformance()
+
+	message := generatePresidentPerformanceMessage(j.ResAllocID, result, j.presidentID, checkRole)
+	broadcastToAllIslands(shared.TeamIDs[j.JudgeID], message)
+
+}
+
+// appointNextPresident returns the island ID of the island appointed to be the president in the next turn
+func (j *judiciary) appointNextPresident(clientIDs []shared.ClientID) shared.ClientID {
+	j.budget -= serviceCharge
+	var election voting.Election
+	election.ProposeElection(baseclient.President, voting.Plurality)
+	election.OpenBallot(clientIDs)
+	election.Vote(iigoClients)
+	return election.CloseBallot()
+}
+
+// generateSpeakerPerformanceMessage generates the appropriate communication required regarding
+// speaker performance to be sent to clients
+func generateSpeakerPerformanceMessage(BID int, result bool, SID shared.ClientID, conductedRole bool) map[int]baseclient.Communication {
```

Comment:
Communication feels like it shouold be in the shared package, isn't it?

Reply:
Agreed and changed